File: baekjoon/StringQuestion/028-1652-누울자리를찾아라.java
import java.io.BufferedReader;
import java.io.InputStreamReader;


public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N, prevIdx, cnt, cntW, cntH;
        N = Integer.parseInt(br.readLine());
        if (N == 1) System.out.print("0 0");
        else {
            char[][] line = new char[N][];

            cntW = 0;
            for (int i = 0; i < N; i++) {
                line[i] = br.readLine().toCharArray();
                cnt = 0; 
                prevIdx = -1;
                for (int j = 0; j < N; j++) {
                    if (line[i][j] == 'X') {
                        if (j - prevIdx > 2) cnt++;
                        prevIdx = j;
                    }
                }
                if (N - prevIdx > 2) cnt++;
                cntW += prevIdx > -1 ? cnt : 1;
            }

            cntH = 0;
            for (int j = 0; j < N; j++) {
                cnt = 0;
                prevIdx = -1;
                for (int i = 0; i < N; i++) {
                    if (line[i][j] == 'X') {
                        if (i - prevIdx > 2) cnt++;
                        prevIdx = i;
                    }
                }
                if (N - prevIdx > 2) cnt++;
                cntH += prevIdx > -1 ? cnt : 1;
            }

            System.out.print(cntW + " " + cntH);
        }
    }
}

File: baekjoon/StringQuestion/028-1652-누울자리를찾아라.py
import sys; input = sys.stdin.readline

N = int(input())
if N == 1: print("0 0")
else:
    metrix = [list(input().rstrip()) for _ in range(N)]

    cntW = 0
    for i in range(N):
        cnt = 0
        prevIdx = -1
        for j in range(N):
            if (metrix[i][j] == 'X'):
                if (j - prevIdx > 2): cnt += 1
                prevIdx = j
        if N - prevIdx > 2: cnt += 1
        cntW += cnt if prevIdx > -1 else 1

    cntH = 0
    for j in range(N):
        cnt = 0
        prevIdx = -1
        for i in range(N):
            if (metrix[i][j] == 'X'):
                if (i - prevIdx > 2): cnt += 1
                prevIdx = i
        if N - prevIdx > 2: cnt += 1
        cntH += cnt if prevIdx > -1 else 1

    print(cntW, cntH)